### ISSUE TITLE
h264 encoding demo

### ISSUE THREAD

**Issue by feuera** [linux-sunxi/cedarx-libs#4]
Is there a working h264 encoding demo? 
enc_dec_demo does not seem to compile. It uses older include files (e.g. #include "encoder_type.h") existing in earlier versions of the "old" directory (I think its renamed to type.h)

There is a demo in old directory called "EncoderDemo" which comiles but seems to have issues with the current kernel/board configurations? My debug output gets flooded with disp pixel sequence and or scaler issues. The Display does not show any valid picture/video.

I use an a13-olinuxino board for testing. 
Thank you, 
Andreas

**Comment by mittorn** [comment]
Use other demo. https://github.com/patrickhwood/h264encoder

**Comment by feuera** [comment]
Thank you very much!
I'll look into that..

**Comment by c1111f1** [comment]
Dear developers,
Thank you for the demo files.
I tried the other demo, but met a problem called Open file /dev/cedar_dev failed.
I used pcDuino board with Allwinner A10 CPU. So could you please explain how to solve this problem?

Thank you very much!

**Comment by oliv3r** [comment]
On 03/22/2014 04:42 PM, c1111f1 wrote:

> Dear developers,
> Thank you for the demo files.
> I tried the other demo, but met a problem called Open file 
> /dev/cedar_dev failed.
> I used pcDuino board with Allwinner A10 CPU. So could you please 
> explain how to solve this problem?
> 
> Did you check permissions or if the file exists at all?

Did you follow the howto? http://linux-sunxi.org/Cedrus

**Comment by c1111f1** [comment]
Thank you for you so fast reply!
I will try it.

**Comment by c1111f1** [comment]
Yes. I used "root" to run that app and it worked. But I still has some problems about color space. I used YUV420 "foreman.yuv" sequence to test the app and used VLC to check the output file. I fount the color was not right but the Y image was OK now. I need to consider the RAW file format.

Thank you again.
